pkg/scheduler/framework/plugins/schedulinggates/scheduling_gates.go

```diff
@@ -53,13 +55,34 @@ func (pl *SchedulingGates) PreEnqueue(ctx context.Context, p *v1.Pod) *framework
 	return framework.NewStatus(framework.UnschedulableAndUnresolvable, fmt.Sprintf("waiting for scheduling gates: %v", gates))
 }
 
-// EventsToRegister returns nil here to indicate that schedulingGates plugin is not
-// interested in any event but its own update.
+// EventsToRegister returns the possible events that may make a Pod
+// failed by this plugin schedulable.
 func (pl *SchedulingGates) EventsToRegister() []framework.ClusterEventWithHint {
-	return nil
+	return []framework.ClusterEventWithHint{
+		// Pods can be more schedulable once it's gates are removed
+		{Event: framework.ClusterEvent{Resource: framework.Pod, ActionType: framework.Update}, QueueingHintFn: pl.isSchedulableAfterPodChange},
```

Comment:
For SchedulingGate, we can register Pod/Update event without QHint because filter out in QHint and PreEnqueue is the same calculation cost - filtering in QHint doesn't add additional value, rather it adds complexity a bit.